### DISCUSSION TITLE
Harder difficulty option for scenarios

### DISCUSSION THREAD

**Discussion by deurklink** [OpenRCT2/OpenRCT2#20175] [Feature requests]
I made this discussion for pull #20170 

I applaud an extra difficulty that tried to combat some easy ways to cheese the game. Right now it's very difficult to design a scenario that cannot be easily cheesed, unless you make it insanely difficult.

Original post from BarryJRowe:

```
This adds a new scenario option under the "Option - Park" dialog, called "Extra difficulty." The idea here is to add some extra checks to discourage some common scenario cheesing techniques. This includes:

1.) Voiding guests now counts as casualties.
2.) Guests will eventually leave queue lines with TVs.
3.) Guests need to ride more rides to stay in the park over time, instead of just 1.
4.) Excessive g-forces or intensity lowers ride value.
5.) Guests will leave the park if they've spent more than 3 times the average guest start cash.
6.) Umbrellas have just an increased max price in rain, not $20.
7.) Advertising guest generation affected by park rating.
8.) Rides with similar stats (length, max speed, etc.) to an existing ride of the same type get value and soft guest cap penalties.

In general, these changes should not affect normal play that much, just optimized play by discouraging coaster spam, death mazes, guest storage, park value bombs, etc. This is still a draft PR, as I have to continue to play test.
```

**Comment by kyphii** [comment]
I've discussed this issue previously in the discord. Here's my thoughts. As I see it, there are three primary balance issues:

**1) Cheese Rides (Microcoasters & Park Value Bombs)**
Ideally the game would encourage players to start with small, compact rides and gradually scale up. Instead, the only reason to build big is if you want to build something cool - It's always better to build small and cheap, since Soft Guest Cap contributions are static and Park Value / Cost ratio is favorable. I see a couple options to address this:
 - Factor in more stats for calculation of SGC and PV ratings. These could include E/I/N, ride duration, etc.
 - Tighten up some stat requirements. A lot of them are very permissive, especially on rides with access to Reverse Shuttle mode. Making them a bit more stringent where possible, such as requiring inversions more often and not negating other requirements when a ride has inversions, could force the player to build better rides.
 - Add a ride duration stat requirement. Most cheese rides are short, and could be countered by this. It would need to be lenient to allow for naturally short rides such as shuttle coasters, so Reverse Shuttle and Powered Launch modes could nullify this requirement.
 - For a positive reinforcement option, introduce a Challenge or Record-Breaking system, similar to TTD's subsidy system. Challenges to build a ride with certain requirements (E/I/N, other stats, ride type, etc.) could occasionally pop up in the news feed, with a time limit to complete them. These could encourage the player to build bigger and better by offering a cash prize, a boost to guest generation, more powerful advertising, a new invention, or some other reward. 

I don't think the highly optimized microcoaster play style should be completely balanced out of existence, so a scenario option for rebalanced difficulty is ideal.

**2) Ride Spam**
The boring but practical way to beat any scenario is to spam microcoasters, one tile mazes, motion simulators, and other cheap, compact rides. The most obvious solution is to make SGC and Park Value contributions fall off with repeat ride types. This should have some leniency, maybe the first 2 or 3 aren't penalized so dueling rides still get a full bonus, but after that diminishing returns should start to kick in. Trying to assess ride similarity as the mentioned PR attempts to do is also an option, though that might be more complicated where a simple solution would work just as well. 

The negatives of this falloff could be offset by introducing bonuses for unique rides. The repeat ride penalty could be lifted for rides that are sufficiently different despite having the same type, such as different operating modes. Maybe we could create a "checklist" of different ride experiences that could be tracked (wooden, multilooper, family, launched, etc). The more that are ticked, the more complete the park's ride lineup. Filling it out gives bonuses to guest generation, guest starting cash, or some other factors. Maybe this could also integrate with a records system as mentioned prior.   

**3) Advertising**
Ads are simply too powerful. They trivialize any scenario, but especially pay-per-entry scenarios, as they pay for themselves extremely quickly. They're also basically required to do well in those scenarios, so it's reduced to either very difficult without ads or Baby Pants Easy with them. The solution might be to introduce more risk/reward with them - Guests spawned by ads might be harder to please, more likely to leave quickly, or less willing to spend money.

**Reply by BarryJRowe** [reply]
1.) EIN ratings might best to avoid for this, as changing the SGC and ride value solves most scenario cheesing (value affects the ride price, and gpc the guest generation). Changing EIN would make sense for cheesing involving the finishing 5 coasters, or create 10 coasters with x excitement goals, but I feel that those aren't really being cheesed now. 

Ride duration as a stat requirement doesn't make sense to me, as many modern rides are designed to be fast but intense. If the duration is short, then the speed/intensity could be required to be high to compensate.

I think challenges are a nice idea, but sound out of scope for this issue and should be a separate feature.

2.) I prefer having a more relaxed requirement than having unique coaster types/modes (eg: at most 2 wooden coasters count to SGC), as there are cases where parks could have a requirement to only build certain coasters types (eg: only wooden, only bobsleds, etc.), and we shouldn't discourage such a scenario if we can. Some key ride features like length, max speed, number of drops, and just a range around the EIN values would do it without much additional work.

**Reply by James103** [reply]
Could there be an additional check to ensure rides with excessive intensity or G-forces etc. don't contribute (as much) to the soft-guest cap, even if the offending ride is the only one of its type?

_Originally posted by @James103 in https://github.com/OpenRCT2/OpenRCT2/pull/20170#discussion_r1189594659_

**Reply by BarryJRowe** [reply]
Makes sense, I can do that.

**Comment by ocalhoun6** [comment]
>2.) Guests will eventually leave queue lines with TVs.

This could still be cheesed with sufficient use of entertainers ... though entertainers are much more expensive.

**Reply by BarryJRowe** [reply]
> > 2.) Guests will eventually leave queue lines with TVs.
> 
> This could still be cheesed with sufficient use of entertainers ... though entertainers are much more expensive.

That's also addressed in the proposed PR - if the time in queue is long enough, then happiness decreases, and if in the queue time pasts a second threshold, guests will leave anyway and become unhappy (for the entertainer case). It took me a while to figure out that entertainers actually decrease the InQueueTime variable for guests standing in line.

**Comment by deurklink** [comment]
My opinion:

1.) Voiding guests now counts as casualties.
- Probably a good change, this would mean a park rating drop if someone just dumps guests into the void. Advertisements ignore park rating for guests generation though as it is now.

2.) Guests will eventually leave queue lines with TVs.
- I like this change. Maybe also for entertainers?

3.) Guests need to ride more rides to stay in the park over time, instead of just 1.
- Don't really have an opinion on this one.

4.) Excessive g-forces or intensity lowers ride value.
- Might be a good way to prevent park value bombs.

5.) Guests will leave the park if they've spent more than 3 times the average guest start cash.
- Sounds fair, it doesn't make ATMs useless, but also stops them from being overly OP.

6.) Umbrellas have just an increased max price in rain, not $20.
- Agree.

7.) Advertising guest generation affected by park rating.
- I agree advertisements are very OP, maybe have it be affected by more than park rating. Maybe a cap on the amount of guests it can generate?

8.) Rides with similar stats (length, max speed, etc.) to an existing ride of the same type get value and soft guest cap penalties.
- To keep it simple, maybe just check for amount of rides of same type?

**Comment by MarcelVos96** [comment]
Having ads being affected by park rating is basically useless as 99% of the times I'm cheesing with ads the park rating is very high anyway. They need to be nerfed in some other way. Maybe halving their guest generation is a good start. 

If it were made so that guests would leave a pay-for-entry scenario quicker then ads would be less necessary in those scenarios in the first place as you're fine without them as long as you have an influx of guests. That would make it more difficult to get to higher guest numbers though, so it might not be ideal, but maybe that could be an interesting part of the difficulty option.

People have already asked me what I think of this and since it will be an opt-in difficulty thing (and not a default rebalance of the game) I'm all for it and it sounds very interesting.

**Reply by James103** [reply]
@ocalhoun6 came up with some other ideas for drawbacks to running ads on a park in https://github.com/OpenRCT2/OpenRCT2/pull/20333#issuecomment-1575685963.

To summarize, one or more of the following could be implemented:
* More expensive ad campaigns.
* Reduced guest happiness for guests attracted by ads.
* Running ads costs park rating, especially in small parks.
* Ads attract less guests with a lower park rating.
* A scenario may have ads unlocked via additional requirements (e.g. reach 8 excitement, open 5 food stalls etc.)

**Comment by BarryJRowe** [comment]
My thoughts so far:

The goal I wanted to address in the PR was primarily around cheesing. The idea was that "normal play" should not be affected by these changes as much as possible, and older scenarios that would be affected can be compatible by just making this a scenario option (all the above changes, with the exception of the advertising change would break [ie: make impossible] existing scenarios I have made for example). As such, a better name for the scenario option might be "Stricter Park Requirements" instead. Thoughts?

As well, changes for this mode don't have to be finalized before release. Since it's an optional scenario mode, it can always be labeled as experimental and await feedback from the community for additional changes until everything is solidified.

Finally, the weakest change was to the advertising. It's harder to come up with a nerf that doesn't change "normal play." Maybe having the campaign guest generation be a low constant amount plus a multiplier of the current natural guest generation rate would be interesting. This would make it so that you get to your SGC faster with ads, but past that the ads are not as effective.

**Reply by ocalhoun6** [reply]
>Maybe having the campaign guest generation be a low constant amount plus a multiplier of the current natural guest generation rate would be interesting. This would make it so that you get to your SGC faster with ads, but past that the ads are not as effective.

Maybe ads should simply act as a multiplier of the natural guest generation? Say, each one multiplies the existing natural guest generation by 1.25 or so. So if you were getting 1000 guests per hour, ads would bump it up to 1250. If you only had 100 guests per hour, it would be bumped up to 125. And if you had 0 guests per hour, it would still be 0.

**Comment by trzejos** [comment]
Could different difficulty rebalancing options be presented as separate scenario options rather than including them all under one extra difficulty flag?

**Comment by ZehMatt** [comment]
I think we should separate two things here, one is the current cheesing/exploiting/glitching whatever term you want to apply to this, the other one is making the game play more challenging which are two different topics. Altering a bunch of systems at the same time is a really bad idea and it will be incredibly hard to balance it all together, turn one knob and something else will be off, we need to focus on single systems and have to see that it will be better with the current content created, trying to attack 10 issues at the same time will lead nowhere. One example would be the guest generation, this is alone is a pretty sensitive and also subjective topic so maybe try to discuss this first and then move on to the next problem and so on. Having so many alterations makes it also incredibly difficult for us to properly review and reason with.

**Comment by BarryJRowe** [comment]
I've made some more changes to the PR based on feedback. I agree with @ZehMatt that this is mainly an anti-cheesing feature. I disagree that these should be separate options however, as that would clutter up the scenario options page. Anti-cheesing is a good umbrella term for what this PR is setting out to do. In that vein, I will rephrase the the proposal in more detail: 

First of all this feature is a "Stricter Park Requirements" scenario option, not anything about "difficulty." For the changes themselves:

1.) Voiding guests now counts as casualties.
- This addresses death mazes and rides that make pay-to-enter parks much easier.

2.) Guests will eventually leave queue lines with TVs.
- This addresses guest storage in queue lines for scenarios with guest goals. It also addresses some monthly income cheesing for "guest dumping."

3.) Guests need to rides more rides to stay in the park over time, instead of just 1.
- This addresses guest storage and monthly income cheesing "guest dumping" where guests with one ride riden are stored in queues/paths and then have the queue/path deleted so they drop down to the path below and then get on rides all at once. The part not addressed in 2.) is when the guests are stored in a benched path (for tiredness) with entertainers around to keep them happy. [Seriously, you can get 4mil in a month with this technique.]

4.) Excessive intensity/g-forces reduce ride value.
- This addresses park value bombs for park value scenarios, as well as park entry pricing cheesing.

5.) [Removed] Guests will leave the park if they've spent more than 3 times the average guest start cash.
- This addresses pay-per-ride parks with the cash machine.

6.) Umbrellas have just an increased max price in rain, not $20.
- Mainly affects monthly shop income, but also general play. 

7.) Advertising guest generation decreases after soft guest cap is reached, stops at 500 above.
- This addresses advertising being overpowered for basically every scenario.

8.) Rides with similar stats (length, max speed, etc.) to an existing ride of the same type get value and soft guest cap penalties.
- This addresses coaster/ride spam (micro or otherwise). This change requires the most playing to verify that it addresses cheesing enough and keeps normal play unaffected. 

9.) Rides must have a least one customer in order to affect the park value and soft guest caps (ride price not affected).
- This is for disconnected ride cheesing, where rides are added to the park just for their park value/soft guest cap calculations and not for any guests to actually ride them. This is a different approach for dealing with park value bombs as well.

10.) Very low velocities + acceleration values do not increase the distance calculation during testing. Negative velocities on chain-lift hills causes trains to stop instead of bounce.
- This is for the block-brakes glitch increasing ride length while the trains are actually stopped, and for using gaps in chain-lift hills to get the train to bounce until another train pushes it through, increasing ride length.

Things that still have to addressed in the PR:

- Play testing (for obvious reasons).
- Some changes like guest leaving queues should be probabilistic for a better feel.
- Performance in large parks - the similarity comparison is a O(n^2) n=number_of_rides check every X seconds for the park value/soft guest cap calculations. That should be fine for 1k rides, but it should be tested and if needed, use a cached value most of the time instead.


As stated, new difficulty options (eg: make a coaster with certain EIN ratings, land purchased is only rented to you, etc.) themselves should be tabled for now. However, I think that community feedback from creators could be useful for getting ideas for what to add, as well as having them create good scenarios to use those new options when they're ready.

**Reply by ocalhoun6** [reply]
> 5.) Guests will leave the park if they've spent more than 3 times the average guest start cash.
> 
>  * This addresses pay-per-ride parks with the cash machine.

I'm really not convinced that this one is necessary. ATMs aren't *that* overpowered. All they do is help you keep the guest count a bit higher and more stable. Without them, broke guests would leave ... which brings you down below your soft guest cap, so new guests with full wallets will quickly come in to replace them. 

(In some ways, going without ATMs can be more beneficial, even in a free entry, pay per ride scenario. For example, each guest will buy only one souvenir and only one ride photo of each type. Going without ATMs -- and therefore having a higher guest turnover -- could actually significantly increase your merchandise sales. Especially relevant in a scenario where shop/stall sales is the scenario goal. The real 'cheesing' here is *not* using ATMs!)

Limiting the effectiveness of ATMs like this could also have a disproportionate effect on larger parks. One great use of ATMs is to make sure the guests still have enough money left to enjoy the park after crossing a huge park and riding lots of rides. Without this, a very large park would probably see the most remote parts of the park practically empty, because by the time guests get that far, they're already broke and decide to go home. With the new save format's increased limits, this is more important than ever.

*Furthermore* ... I'm not aware of any ATM strategy that I could really call 'cheesing'. Certainly not on the level of the other things you're addressing here. Yes, it's to your benefit to place plenty of ATMs, but it's not as if anybody is going around blowing past scenario goals simply by using ATMs in a clever way or using a lot of them. Unlike most of the other things you're talking about here, ATMs alone won't guarantee a scenario win. You can't make a difficult scenario trivial just by using ATMs.

**Reply by James103** [reply]
A more realistic way of handling this is to have each guest keep track of two money values: one for the amount of cash the guest is currently carrying and another for how much they have in their bank account. When a guest uses an ATM, the amount of cash withdrawn is debited from that guest's bank account; if their bank balance is low, the guest leaves the park soon after.

The initial amount of money in each guest's bank account could be derived in two ways:
1. Multiply the average amount of cash each guest starts with by a constant (such as 3).
2. Roll a random variable, the parameters of which are defined in the scenario options, and use the result directly.

**Reply by ZehMatt** [reply]
> I disagree that these should be separate options however

I may have not properly addressed that in my comment, I'm not talking about extra options here but rather about tackling problems one by one, the same goes for discussions about each specific problem. As you can see people throw around a lot of suggestions and ideas which is fine but since its not really single topic it will typically lead to just having lots of comments without a clear direction at the end. To recap on what I said, it would be best to discuss each problem individually and also address those problems individually. We can definitely do bigger PRs but it would help to already know what the problem is and how to address them beforehand. Apologies if my initial comment wasn't clear.

**Reply by kyphii** [reply]
I don't think the goal should be to make the game more difficult per se, ideally it should be about the same. I think the goal should be to make tweaks that make playing optimally closer to the intended experience. 

That'll mean not just nerfing the existing strategies, but adding buffs that make the alternative about as viable. This could be anything from minor tweaks to whole new systems that provide positive feedback to the player. The changes currently in the PR all negatively affect the player, so maybe some positive tweaks to promote a better playstyle should be considered.

Regarding the current proposed changes in the PR:
1-3) Make sense.
4) I'm not convinced on this solution, I think having things like ride duration and excessive G forces makes more sense.
5-6) Make sense.
7) As I mentioned before, I think it might be better to instead introduce some more risk to ads' reward, rather than flatout nerfing them.
8) I think this is overly complex where a simple check for ride type equivalence + operating mode would be sufficient. Anything complex here is likely to cause confusion since the system is hidden from the player.
9-10) Make sense, I'd have to see the implementation to really get a sense for it though.

**Reply by ocalhoun6** [reply]
>A more realistic way of handling this is to have each guest keep track of two money values:

Well, that depends what the goal here is.

If the goal is to make the ATM function *more realistic*, sure -- it would be good to talk about stuff like this.

But if the goal is to be an anti-cheesing feature, then I don't think any modification to ATM function is necessary. Because ATMs aren't really cheesing to begin with.

Yes, placing lots of ATMs can help you win scenarios ... but so can placing lots of, say, handymen. So are we going to say that using handymen is cheesing now?

My point is that for all the other cheesing methods addressed here, doing that alone can trivialize beating a scenario, at least in certain scenario goals -- microcoaster spam, guest storage in ultra-long queues, guest dumping from such a queue to meet monthly ride income goals, $20 umbrellas + rain to meet shop sale goals, overpowered ads to meet guest count goals, etc, etc, etc. *Except* for the ATMs. There's no situation where you can make an otherwise difficult scenario trivial just by using ATMs.

So if we want to look at making ATMs more realistic, sure ... we could talk about that. But it doesn't fit in an anti-cheesing feature.

**Reply by BarryJRowe** [reply]
> 4) I'm not convinced on this solution, I think having things like ride duration and excessive G forces makes more sense. 

I forgot to update that entry. It's uses excessive g forces and intensity now instead. I've edited the original post.

**Comment by BarryJRowe** [comment]
I've edited my previous comment with some changes to the existing PR. For the two newer additions, I've copied them below:

9.) Rides must have a least one customer in order to affect the park value and soft guest caps (ride price not affected).
- This is for disconnected ride cheesing, where rides are added to the park just for their park value/soft guest cap calculations and not for any guests to actually ride them. This is a different approach for dealing with park value bombs as well.

10.) Very low velocities + acceleration values do not increase the distance calculation during testing. Negative velocities on chain-lift hills causes trains to stop instead of bounce.
- This is for the block-brakes glitch increasing ride length while the trains are actually stopped, and for using gaps in chain-lift hills to get the train to bounce until another train pushes it through, increasing ride length.


I think the low distance check is fine for catching this case, though I'm not so sure about my implementation for the chain-lift hill case.


I think what's left now is for more clarity on what to do about ad effectiveness, smoothing out some probabilities for guests leaving queue lines, and reviewing the specific implementations.

**Reply by MarcelVos96** [reply]
> 9.) Rides must have a least one customer in order to affect the park value and soft guest caps (ride price not affected).
> This is for disconnected ride cheesing, where rides are added to the park just for their park value/soft guest cap calculations and not for any guests to actually ride them. This is a different approach for dealing with park value bombs as well.

This will cause issues in regular play. Imagine a scenario with a long pathway from the map edge to the park entrance (like Mystic Mountain). If you can't increase the soft guest cap before any guests have ridden the rides, the first guests have to be generated on a soft guest cap of zero, which is four times slower (or 16 times with harder guest generation). They then take a while to even reach the park and possibly a little while longer to ride the rides and finally increase the cap. The guest generation is already slow at the start as your park rating is low, and this would make it even slower.

**Reply by BarryJRowe** [reply]
Fair enough. I've changed this to only come into affect after 500 guests are in the park.

**Comment by BarryJRowe** [comment]
I've done a minor change to make exiting queue line with TVs in them to be a little random after the time limit is reached.

With that, are there any other issues to address? Everything brought up seems to be addressed from what I can see.

**Reply by ocalhoun6** [reply]
I still don't see why the ATM change is included in this. Because there's no way to 'cheese' a scenario just by using ATMs, unlike *everything* else you're addressing.

Every other thing you've done is to address some shortcut that would make certain types of scenarios trivial to win. But there's no situation in which ATMs alone make a scenario trivial. In some cases (pay for entry parks that are short on cash flow) you're actually better off not using them, even in vanilla.

**Reply by BarryJRowe** [reply]
The ATM part is already removed from the PR.

**Reply by ocalhoun6** [reply]
Oh, sorry. I must have missed that.
